duckduckgo/chrome-zeroclickinfo
# Add ToSDR to privacy grade

**Reviewer:**



## Description:

I gathered all of the ToSDR data using `scripts/tosdr.js` and processed into our own internal data structure that we will use to adjust the privacy grade.
https://github.com/duckduckgo/chrome-zeroclickinfo/compare/beta...jd/tosdr?expand=1#diff-b5b236475963defb1316e2989db286f6

The overall adjustment to the grade is pre-calculated and includes a `match` object that lists the good and bad reasons for the grade. 

I manually built a list of all of the good and bad privacy sounding topics to use for adjusting the grade. 
https://github.com/duckduckgo/chrome-zeroclickinfo/compare/beta...jd/tosdr?expand=1#diff-26cd4543b120a64e612a321d503a8996

When grading a site there is a new tosdr property that contains the overall score and reasons for the adjustment. 

<img width="740" alt="screen shot 2017-08-31 at 4 50 39 pm" src="https://user-images.githubusercontent.com/1882892/29948780-0d8ad700-8e6e-11e7-8dad-a8cf05cc8e05.png">


## Steps to test this PR:

- test http://mint.com before loading this PR. It should be a "C"
- load this PR and visit the site again. It's now a "D" since it has one bad ToSDR topic.
https://github.com/duckduckgo/chrome-zeroclickinfo/compare/beta...jd/tosdr?expand=1#diff-b5b236475963defb1316e2989db286f6R452

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
